types/react/v16/test/managedAttributes.tsx

```diff
@@ -1,276 +1,261 @@
 
-// const MemoFunctionalComponent = React.memo(FunctionalComponent);
-// const MemoAnnotatedDefaultProps = React.memo(AnnotatedDefaultProps);
-// const LazyMemoFunctionalComponent = React.lazy(async () => ({ default: MemoFunctionalComponent }));
-// const LazyMemoAnnotatedDefaultProps = React.lazy(async () => ({ default: MemoAnnotatedDefaultProps }));
+const memoTests = [
+    // $ExpectError
+    <MemoFunctionalComponent />,
+    // $ExpectError won't work as long as FunctionalComponent doesn't work either
+    <MemoFunctionalComponent str='abc' />,
+    // $ExpectError
+    <MemoAnnotatedDefaultProps />,
+    <AnnotatedDefaultProps str='abc' />,
+    // $ExpectError this doesn't work despite JSX.LibraryManagedAttributes returning the correct type
+    <MemoAnnotatedDefaultProps str='abc' />,
+    // $ExpectError won't work as long as FunctionalComponent doesn't work either
+    <LazyMemoFunctionalComponent str='abc' />,
+    // $ExpectError
+    <LazyMemoAnnotatedDefaultProps />,
+    // $ExpectError this doesn't work despite JSX.LibraryManagedAttributes returning the correct type
+    <LazyMemoAnnotatedDefaultProps str='abc' />
+];
 
-// const memoTests = [
-//     // $ExpectError
-//     <MemoFunctionalComponent />,
-//     // $ExpectError won't work as long as FunctionalComponent doesn't work either
-//     <MemoFunctionalComponent str='abc' />,
-//     // $ExpectError
-//     <MemoAnnotatedDefaultProps />,
-//     <AnnotatedDefaultProps str='abc' />,
-//     // $ExpectError this doesn't work despite JSX.LibraryManagedAttributes returning the correct type
-//     <MemoAnnotatedDefaultProps str='abc' />,
-//     // $ExpectError won't work as long as FunctionalComponent doesn't work either
-//     <LazyMemoFunctionalComponent str='abc' />,
-//     // $ExpectError
-//     <LazyMemoAnnotatedDefaultProps />,
-//     // $ExpectError this doesn't work despite JSX.LibraryManagedAttributes returning the correct type
-//     <LazyMemoAnnotatedDefaultProps str='abc' />
-// ];
+const ForwardRef = React.forwardRef((props: Props, ref: React.Ref<ComponentWithNoDefaultProps>) => (
+    <ComponentWithNoDefaultProps ref={ref} {...props}/>
+));
+ForwardRef.defaultProps = defaultProps;
 
-// type AnnotatedDefaultPropsLibraryManagedAttributes = JSX.LibraryManagedAttributes<typeof AnnotatedDefaultProps, Props>;
-// // $ExpectType AnnotatedDefaultPropsLibraryManagedAttributes
-// type FunctionalComponentLibraryManagedAttributes = JSX.LibraryManagedAttributes<typeof FunctionalComponent, Props>;
-// // $ExpectType FunctionalComponentLibraryManagedAttributes
-// type MemoFunctionalComponentLibraryManagedAttributes = JSX.LibraryManagedAttributes<typeof MemoFunctionalComponent, Props>;
-// // $ExpectType FunctionalComponentLibraryManagedAttributes
-// type LazyMemoFunctionalComponentLibraryManagedAttributes = JSX.LibraryManagedAttributes<typeof LazyMemoFunctionalComponent, Props>;
```

Comment:
I removed these tests because the string representation of the types vary based on the TS version. I think the behavioral tests should be sufficient.